subsys/net/openthread/rpc/server/ot_rpc_udp_server.c

```diff
@@ -0,0 +1,390 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <nrf_rpc/nrf_rpc_serialize.h>
+#include <ot_rpc_ids.h>
+#include <ot_rpc_types.h>
+#include <ot_rpc_common.h>
+
+#include <nrf_rpc_cbor.h>
+
+#include <openthread/error.h>
+#include <openthread/instance.h>
+
+#include <openthread/udp.h>
+
+#include <zephyr/net/openthread.h>
+#include <zcbor_decode.h>
+#include <string.h>
+
+#include "ot_rpc_message_server.h"
+
+#define OT_UDP_SOCKETS 8
+
+typedef struct nrfUdpSocket {
+	ot_socket_key mKey;
+	otUdpSocket mSocket;
+} nrfUdpSocket;
+
+static nrfUdpSocket sSockets[OT_UDP_SOCKETS];
+
+static nrfUdpSocket *nrf_udp_find_socket(ot_socket_key key)
+{
+	for (size_t i = 0; i < OT_UDP_SOCKETS; i++) {
+		if (key == sSockets[i].mKey) {
+			return &sSockets[i];
+		}
+	}
+
+	return NULL;
+}
+
+static nrfUdpSocket *nrf_udp_alloc_socket(ot_socket_key key)
+{
+	nrfUdpSocket *ret = nrf_udp_find_socket(key);
+
+	if (ret != NULL) {
+		return ret;
+	}
+
+	ret = nrf_udp_find_socket(0);
+
+	if (ret != NULL) {
+		memset(ret, 0, sizeof(nrfUdpSocket));
```

Comment:
Why is it required? It should be zero if it is empty. Otherwise, searching by 0 doesn't have sense.

Reply:
Yo are correct removed the code.